skt/runner.py

```diff
@@ -217,9 +217,14 @@ def getresultstree(cls, taskspec):
         args = ["bkr", "job-results", "--prettyxml", taskspec]
 
         err_strings = ["ProtocolError", "503 Service Unavailable"]
-        stdout, _, _ = retry_safe_popen(err_strings, args,
-                                        stderr=subprocess.PIPE,
-                                        stdout=subprocess.PIPE)
+        stdout, stderr, returncode = retry_safe_popen(err_strings, args,
+                                                      stderr=subprocess.PIPE,
+                                                      stdout=subprocess.PIPE)
+
+        if returncode:
+            logging.warning(stdout)
+            logging.warning(stderr)
+            raise RuntimeError('failed getting Beaker job-results')
```

Comment:
Can you explain the reasoning for the exception in the commit message please? It only mentions the added logging but not why we should immediately abort if something goes wrong so the functionality changes don't match.

Reply:
I've expanded the description in the commit message. Thanks. :)

Reply:
Thanks, that makes more sense now! And creates a followup question: do we also want to remove the beaker-results xml creation form here, since we've 1) removed them as artifacts and 2) even when they were artifacted we redownloaded the results from beaker in the pipeline?

Feel free to merge this PR if you don't want to make that followup cleanup commit.

Reply:
Nice catch, that file isn't used anymore. So let's not keep it.